src/WebformBuilder.js

```diff
@@ -959,15 +959,16 @@ export default class WebformBuilder extends Component {
       return;
     }
     let remove = true;
-    if (
-      !component.skipRemoveConfirm &&
+    const removingComponentsGroup = !component.skipRemoveConfirm &&
       (
         (Array.isArray(component.components) && component.components.length) ||
         (Array.isArray(component.rows) && component.rows.length) ||
         (Array.isArray(component.columns) && component.columns.length)
-      )
-    ) {
-      const message = 'Removing this component will also remove all of its children. Are you sure you want to do this?';
+      );
+
+    if (this.options.askForComponentsRemoval || removingComponentsGroup) {
```

Comment:
This seems like the incorrect logic...  

If ```askForComponentsRemoval``` is true, it will always ask for confirmation, and if it is false, then it will work like it always has where it will only ask if the skipRemoveConfirm is set to false. Seems like inverted from what you want.  

I think the following is what you are after...

```
if (!this.options.skipRemoveConfirm && removingComponentsGroup) {
```

This way it is reverse compatible, and will only skip if you provide a ```skipRemoveConfirm``` within the options. Your thoughts?

Reply:
I think what you proposed in otherwise of what I need 😉
We want a possibility to **always ask before removing a component**. In the solution you proposed, the only change would be that I can pass `skipRemoveConfirm` in options which - when set - would make it NOT asking for the component's removal. The goal is otherwise - to ALWAYS ask for components removal, independently whether it's a group of components or a single component.

 The previous behavior was that we only ask for removal if the components group is removed and `skipRemoveConfirm` is not defined or false in the component's options (`component.skipRemoveConfirm`).  wanted to keep it backwards-compatible, so I introduced a new option called `this.options.askForComponentsRemoval`. When this is set to true, it will always ask before removing any component. If this is not set (the default, current behavior) it works as previously (now).

That's why you have this line:
`if (this.options.askForComponentsRemoval || removingComponentsGroup) {`
because it will ask for components removal in one of two cases:

- `removingComponentsGroup` is true, which represents the old (current) conditions

- `this.options.askForComponentsRemoval` which is a newly added option that can be provided to the builder

I guess it clarifies, but let me know if we're still not on the same page 🙂

Reply:
@travist are we on the same page here?

Reply:
I will need to run this locally to verify that it is reverse compatible. At first glance, this looks like it would change behavior on forms that were built without this change. If that is the case, then we will not be able to take this "as is".

Reply:
I don't think it will affect the forms build without this change. As far as I tested, the code without a new option should work as it used to.
Basically the previous condition (`!component.skipRemoveConfirm && (... arrays checking code..)` ) has been placed within `removingComponentsGroup` variable. If this variable is set, the dialog is displayed with the same message as previously. If it's not true, the dialog is skipped. Just as it was before.
But of course run your tests and let me know.

Reply:
yes you are right. Looks good.

Reply:
Maybe you really meant to make the variable say... ```alwaysConfirmRemoval```

Reply:
Ah, now I see what you mean. My intention adding this feature was to add one, new possibility - to allow **asking for every component's removal**. I assumed that we should **always ask for grouping components' removal** (columns, fieldset etc.), because its been implemented like that.
With these assumptions, **if you do not set `askForComponentsRemoval` at all or set it to `false` you'd get the old (current) behavior**.
And exactly, maybe just the naming is an issue here. I see two options:

1. Rename variable `askForComponentsRemoval` to `alwaysConfirmRemoval` or maybe `askForAllComponentsRemoval`
2. Implement the other behavior you assumed could be possible, i.e. not asking for ANY components removal (neither single components nor grouping ones) when `askForComponentsRemoval` is `false`.

To be honest, from the functional point of view, I think we should keep to always ask before removing a grouping component. I wouldn't let disabling it with an option. If the user has many components grouped in a grouping component it would be very frustrating to accidentally delete it all. What do you think?

Reply:
I am alright changing the variable name so it is more descriptive of what is going on. We could be overly verbose and make it ```alwaysConfirmComponentRemoval```

Reply:
Done and committed 😉